### DISCUSSION TITLE
Can I code simple pixel-based graphics/"demos" in Fyrox?

### DISCUSSION THREAD

**Discussion by akavel** [FyroxEngine/Fyrox#695] [Q&A]
I want to have some small fun with simple pixel-based graphics/"generative art". E.g. in the style of Love2d Lua engine, or "gameboy-like", etc. At the moment I don't really want to write full-blown "games", but just graphics toys. I also don't know at all any shader language. Is it possible to do such stuff in Fyrox? Notably, I would like it to have big, visible pixels, even on modern high-DPI screens, to display huge blocky pixels, like an old 320x200 resolution, or 640x480, or gameboy, etc. Can I somehow do it fairly easily in Fyrox? Even more ideally, with some hot-reload action... but I can probably live without this if impossible. Like for example, playing with the "Conway's Game of Life" coded on my own, or Mandelbrot fractal, etc.

*edit:* maybe somehow using the "pixels" crate (https://github.com/parasyte/pixels) in Fyrox with the builtin editor?

**Comment by mrDIMAS** [comment]
Sorry, I saw your question, but I got distracted and forgot about it. To your question: in short - you can, all you need to do is set orthographic projection box of a camera to your desired screen resolution. This way pixel size rendered by the camera will be equal to physical pixel on the screen. Next you need to specify proper filtering for all textures to use nearest filtration to remove smoothing. The final step is to simply draw sprites on screen with these textures. You can freely adjust these parameters to find the most suitable. Hope this helps.
